### ISSUE TITLE
[1.16.5-0.2.3] Flywheel makes Bumblezone's Sugar Water/Honey Fluid rendering break when in the fluid

### ISSUE THREAD

**Issue by TelepathicGrunt** [Engine-Room/Flywheel#38]
Hello! I am the developer of The Bumblezone mod: https://www.curseforge.com/minecraft/mc-mods/the-bumblezone-forge

I have received a report about a modpack making swimming in my Sugar Water Fluid just turn the screen full yellow and unable to see. So after some testing, I narrowed it down to Flywheel and Create. So knowing Flywheel is a rendering based mod, I decided to make the report here instead of Create's. Here's the log and pictures of the issue:
Latest.log: https://paste.ee/p/tz00L
![image](https://user-images.githubusercontent.com/40846040/131428737-d22d3f50-ba9b-4f7b-ba37-80a91f8e6b31.png)
![image](https://user-images.githubusercontent.com/40846040/131429721-e47c6f89-a13b-4393-b674-e4d78d46a6b8.png)

And here's what it is suppose to look like:
![image](https://user-images.githubusercontent.com/40846040/131428956-40c4cbb9-2de0-4825-90a7-181f45c5d530.png)

The way my sugar water works is that it is basically water with a lighter color texture and a secondary very faint texture overlaid on top of it to make faint sugar marks on the screen. The dimension's water fog color is set to 11700268 so that the water and my water are both yellowish in my dimension.

I did further testing and vanilla water placed in my dimension renders just fine when in it. If I take my Sugar Water Block to the overworld and go into it, it is still a yellow screen which means it isn't my biome's water fog color that is the issue. Most likely, it is probably my sugar water overlay code or something else specific to my block that Flywheel doesn't seem to like. Here's some of my code if it helps:

Sugar water overlay rendering code:
https://github.com/TelepathicGrunt/Bumblezone/blob/latest-released/src/main/java/com/telepathicgrunt/the_bumblezone/client/rendering/FluidRender.java

Sugar water fluid class:
https://github.com/TelepathicGrunt/Bumblezone/blob/latest-released/src/main/java/com/telepathicgrunt/the_bumblezone/fluids/SugarWaterFluid.java

I hope this helps and let me know if there's anything I need to fix on my end to fix this incompat!

**Comment by Jozufozu** [comment]
Thanks for the report! This is a weird issue for sure. Initially I though it might be a depth buffer related issue, but my testing has shown that isn't it. There's likely some GL state flywheel isn't correctly tearing down. I'm super busy until Friday, but this weekend I'll be able to look into this further.

**Comment by TelepathicGrunt** [comment]
I just updated Bumblezone to have Honey Fluid and it appears to have the same issue with Flywheel. If it is any help, I did a test build of bumblezone with my RenderBlockOverlayEvent events commented out but no change. I'm not sure where the yellow fig is coming from.
[the_bumblezone_forge-2.5.0-test+1.16.5.zip](https://github.com/Jozufozu/Flywheel/files/7274580/the_bumblezone_forge-2.5.0-test%2B1.16.5.zip)

I'll keep poking around and see what feature from my mod could be triggering this

**Comment by TelepathicGrunt** [comment]
After more testing, I am stumped now. Just fyi, the Honey Fluid is not tagged as water or lava. 
I disabled the RenderBlockOverlayEvent events, commented out my water fog color mixin for honey fluid, changed all the biome's colors so nothing is yellow (water color and water fog color for biome json), changed the dimension's type to the nether's so it uses the nether's DimensionSpecialEffects. And yet, going in either Sugar Water or Honey Fluid gives a yellow dense fog that blinds the player.  Here's the jar with as much stuff disabled or changed as I can in case it helps:
[the_bumblezone_forge-2.5.0-test2+1.16.5.zip](https://github.com/Jozufozu/Flywheel/files/7274603/the_bumblezone_forge-2.5.0-test2%2B1.16.5.zip)

So yeah, I got nothing and no idea what it is about my fluids that is causing this. Let me know if you figure it out and I'll fix the issue if is on my end.

**Comment by TelepathicGrunt** [comment]
i figured out what exactly about my fluid was causing the incompat. it was this method...
https://github.com/TelepathicGrunt/Bumblezone/blob/dadad5dca80b4d1738d1688db09efb45e9a7e6d9/src/main/java/com/telepathicgrunt/the_bumblezone/fluids/TestFluid.java#L15-L18

I made a new branch and stripped everything from my mod but the sugar water fluid. then I made a new fluid from scratch and slowly added the sugar water stuff until it caused the yellow screen incompat. That method + [a mixin into the same method](https://github.com/TelepathicGrunt/Bumblezone/blob/1.16/src/main/java/com/telepathicgrunt/the_bumblezone/mixin/blocks/WaterEquivalenceMixin.java) on vanilla water was to make the sugar water and vanilla water connect to each other and not show a wall between them. That way they appear as a single fluid rather than two separate fluid with a dip between them and a wall of falling fluid texture
https://i.imgur.com/JTKuusS.mp4 

Note: even without the mixin for the vanilla fluid, just overriding that method just on the modded fluid is enough to trigger this bug
